src/net/sourceforge/kolmafia/KoLCharacter.java

```diff
@@ -2326,6 +2327,32 @@ public static final boolean hasGs(String name) {
     return name.contains("g") || name.contains("G");
   }
 
+  private static final Pattern I_PATTERN = Pattern.compile("[Ii]");
+
+  public static final int getLetterIs(String name) {
+    return (int) KoLCharacter.I_PATTERN.matcher(name).results().count();
+  }
```

Comment:
I guess. Seems like you could stream over the characters in the string, somehow, and count Is, rather than using Patterns and Matchers.

Reply:
this is how we do it for Bs (`getBeeosity` above that in the file). The `contains` approach has a comment mentioning that it's a perf improvement; not sure how true that is.

I also wrote `LetterBonus.java` for Maximizer.

Reply:
So I (finally) ran the numbers on doing this

Turns out the `contains` comment about the perf improvement is true - I suspect the JVM is actually offloading the `indexOf` check to a native implementation explaining the two passes being quicker than looping over once in plain Java.

The stream based approach is fairly slow - I suspect that's just due to the overhead of the stream mechanics internally.

The fastest count implementation was a simple for loop testing each character 🤷‍♂️

https://github.com/flaminscotsman/kolmafia-stringutils-performance if you want the full numbers

I can swap the implementation(s) as part of this PR or just change the `eyes` and `ewes` implementations now or change them all in a follow up PR?

Also, not sure if `KolCharacter` is the nicest place for them? Could centralize it all in a `LetterUtils` or something!

Reply:
This is fine. Perhaps we could move {Beeosity, Eweosity, and Eyeosity} methods to a generalized LetterUtils package, by and by, if we care enough,  but it need not be in this PR.